src/mlpack/core/optimizers/cmaes/cmaes_impl.hpp

```diff
@@ -0,0 +1,765 @@
+            << stopTolUpXFactor << ", larger initial standard deviation recommended."
+            << std::endl;
+            end = true;
+        break;
+      }
+    }
+
+    // Condition of C greater than dMaxSignifKond
+    if (maxEW >= minEW* dMaxSignifKond)
+    {
+       Log::Info << "ConditionNumber: maximal condition number " << dMaxSignifKond
+          << " reached. maxEW=" << maxEW <<  ",minEW=" << minEW << ",maxdiagC="
+          << maxdiagC << ",mindiagC=" << mindiagC << std::endl;
+        end = true;
+    }
+
+    // Principal axis i has no effect on xmean, ie. x == x + 0.1* sigma* rgD[i]* B[i]
+    if (!diag)
+    {
+      for (iAchse = 0; iAchse < N; ++iAchse)
+      {
+        fac = 0.1* sigma* rgD[iAchse];
+        for (iKoo = 0; iKoo < N; ++iKoo)
+        {
+          if (xmean[iKoo] != xmean[iKoo] + fac* B(iKoo,iAchse))
+            break;
+        }
+        if (iKoo == N)
+        {
+           Log::Info << "NoEffectAxis: standard deviation 0.1*" << (fac / 0.1)
+              << " in principal axis " << iAchse << " without effect" << std::endl;
+           end = true;
+          break;
+        }
+      }
+    }
+    // Component of xmean is not changed anymore
+    for (iKoo = 0; iKoo < N; ++iKoo)
+    {
+      if (xmean[iKoo] == xmean[iKoo] + sigma*std::sqrt( C(iKoo,iKoo) )/double(5))
+      {
+         Log::Info << "NoEffectCoordinate: standard deviation 0.2*"
+            << (sigma*std::sqrt( C(iKoo , iKoo) ) ) << " in coordinate " << iKoo
+            << " without effect" << std::endl;
+        end = true;
+        break;
+      }
+    }
+
+    if (countevals >= stopMaxFunEvals)
+    {
+       Log::Info << "MaxFunEvals: conducted function evaluations " << countevals
+          << " >= " << stopMaxFunEvals << std::endl;
+       end = true;
+    }
+    if (gen >= stopMaxIter)
+    {
+       Log::Info << "MaxIter: number of iterations " << gen << " >= "
+          << stopMaxIter << std::endl;
+      end = true;
+    }
+
+    return end;
+  }
+
+  /**
+   * Conducts the eigendecomposition of C into B and D such that
+   * \f$C = B \cdot D \cdot D \cdot B^T\f$ and \f$B \cdot B^T = I\f$
+   * and D diagonal and positive.
+   * @param force For force == true the eigendecomposion is conducted even if
+   *              eigenvector and values seem to be up to date.
+   */
+   template<typename funcType>
+   void CMAES<funcType>::updateEigensystem(bool force)
+  {
+    if (!force)
+    {
+      if (eigensysIsUptodate)
+        return;
+      // return on modulo generation number
+      if (gen < genOfEigensysUpdate + updateCmode.modulo)
+        return;
+    }
+
+     if (!arma::eig_sym(rgD, B, C))
+        assert("eigen decomposition failed in neuro_cmaes::eigen()");
+
+    // find largest and smallest eigenvalue, they are supposed to be sorted anyway
+    minEW = rgD.min();
+    maxEW = rgD.max();
+
+    if (doCheckEigen) // needs O(n^3)! writes, in case, error message in error file
```

Comment:
this was made to check whether eigen decomposition happened correctly or not .
since now we are using the eigen function i removed the function and its implementation as well.
Instead used the return type bool given by arma::eigen() that will itself tell if decomposition happened or not correctly.
thanks 
i thought to just keep this as an extra check of safety but its not actually that much required ..
thanks @zoq